gravitee-apim-portal-webui-next/src/app/applications/application/application-tab-logs/application-tab-logs.component.ts

```diff
@@ -13,13 +13,127 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { Component } from '@angular/core';
+import { AsyncPipe, DatePipe } from '@angular/common';
+import { Component, computed, Input, OnInit, Signal, signal, WritableSignal } from '@angular/core';
+import { toObservable } from '@angular/core/rxjs-interop';
+import { MatButton, MatIconButton } from '@angular/material/button';
+import { MatIcon } from '@angular/material/icon';
+import {
+  MatCell,
+  MatCellDef,
+  MatColumnDef,
+  MatHeaderCell,
+  MatHeaderCellDef,
+  MatHeaderRow,
+  MatHeaderRowDef,
+  MatRow,
+  MatRowDef,
+  MatTable,
+} from '@angular/material/table';
+import { catchError, map, Observable, switchMap, tap } from 'rxjs';
+import { of } from 'rxjs/internal/observable/of';
+
+import { LoaderComponent } from '../../../../components/loader/loader.component';
+import { Application } from '../../../../entities/application/application';
+import { LogsResponseMetadataApi, LogsResponseMetadataTotalData } from '../../../../entities/log/log';
+import { CapitalizeFirstPipe } from '../../../../pipe/capitalize-first.pipe';
+import { ApplicationLogService } from '../../../../services/application-log.service';
+
+interface LogVM {
+  apiName: string;
+  apiVersion: string;
+  timestamp: number;
+  method: string;
+  status: number;
+}
 
 @Component({
   selector: 'app-application-tab-logs',
   standalone: true,
-  imports: [],
+  imports: [
+    AsyncPipe,
+    LoaderComponent,
+    CapitalizeFirstPipe,
+    MatCell,
+    MatCellDef,
+    MatColumnDef,
+    MatHeaderCell,
+    MatHeaderRow,
+    MatHeaderRowDef,
+    MatIcon,
+    MatRow,
+    MatRowDef,
+    MatTable,
+    MatHeaderCellDef,
+    DatePipe,
+    MatIconButton,
+    MatButton,
+  ],
   templateUrl: './application-tab-logs.component.html',
   styleUrl: './application-tab-logs.component.scss',
 })
-export class ApplicationTabLogsComponent {}
+export class ApplicationTabLogsComponent implements OnInit {
+  @Input()
+  application!: Application;
+
+  logs$: Observable<LogVM[]> = of([]);
+
+  pagination: Signal<{ hasPreviousPage: boolean; hasNextPage: boolean; currentPage: number; totalPages: number }> = computed(() => {
+    const totalPages = Math.ceil(this.totalLogs() / 10);
+
+    return {
+      hasPreviousPage: this.currentLogsPage() > 1,
+      hasNextPage: this.currentLogsPage() < totalPages,
+      currentPage: this.currentLogsPage(),
+      totalPages,
+    };
+  });
+
+  displayedColumns: string[] = ['api', 'timestamp', 'httpMethod', 'responseStatus'];
+
+  private currentLogsPage: WritableSignal<number> = signal(1);
```

Comment:
Should we read the `page` value from a query param in the activated route?

Reply:
Yes. I'll do that refactor.